elyra/pipeline/component_catalog.py

```diff
@@ -44,40 +48,144 @@
 
 
 class CacheUpdateManagerThread(Thread):
-    def __init__(self, log: Logger, component_cache: Dict[str, Dict], cache_task_queue: Queue):
+    def __init__(self,
+                 log: Logger,
+                 component_cache: Dict[str, Dict],
+                 cache_queue: Queue,
+                 manifest_queue: Queue,
+                 manifest_filename: str):
         super().__init__()
 
         self.setDaemon(True)
 
         self.log = log
         self._component_cache = component_cache
-        self._cache_task_queue = cache_task_queue
+        self._cache_queue = cache_queue
+        self._manifest_queue = manifest_queue
+        self._manifest_filename = manifest_filename
 
         self._threads: List[CacheUpdateThread] = []
 
         self.stop_event = Event()
 
     def run(self):
-        outstanding_threads = False
-
         while not self.stop_event.is_set():
-            timeout = NONBLOCKING_TIMEOUT if outstanding_threads else BLOCKING_TIMEOUT
             try:
-                # Get a task from the queue
-                catalog, operation = self._cache_task_queue.get(timeout=timeout)
-
+                # Get a task from the manifest queue
+                catalog, action = self._manifest_queue.get(timeout=NONBLOCKING_TIMEOUT)
             except Empty:
-                # No task exists in the queue, proceed to check for thread execution
+                # No task exists in the manifest queue, proceed to check for cache tasks
                 pass
-
             else:
-                # Create and start a thread for the task
-                updater_thread = CacheUpdateThread(self._component_cache, self._cache_task_queue, catalog, operation)
-                updater_thread.start()
+                # Make the required update to the manifest file
+                if os.path.isfile(self._manifest_filename):
+                    with open(self._manifest_filename, 'r') as f:
+                        manifest = json.loads(f.read())
+
+                write_required = False
+                if action == 'cache':
```

Comment:
Per offline discussion, we'll keep this variable named `action` because the subsequent `if` block acts on `cache_action`s and the `else` block acts on `manifest_action`s